gc/base/MemorySubSpaceSemiSpace.cpp

```diff
@@ -970,37 +973,49 @@ MM_MemorySubSpaceSemiSpace::checkSubSpaceMemoryPostCollectResize(MM_EnvironmentB
 				extensions->heap->getResizeStats()->setLastExpandReason(SCAV_RATIO_TOO_HIGH);
 			}
 
-			/* If the average scavenge to interval ratio is less than the minimum, try to contract */
-			if(_averageScavengeTimeRatio < extensions->dnssExpectedTimeRatioMinimum
-					&& (NULL != _physicalSubArena) && _physicalSubArena->canContract(env) && (maxContractionInSpace(env) != 0)) {
-				double desiredContractionFactor, adjustedContractionFactor;
+			if((NULL != _physicalSubArena) && _physicalSubArena->canContract(env) && (maxContractionInSpace(env) != 0)) {
+				if (_averageScavengeTimeRatio < extensions->dnssExpectedTimeRatioMinimum) {
+					/* If the average scavenge to interval ratio is less than the minimum, try to contract */
 
-				/* Try to reach 200% of the expected minimum time ratio through contraction */
-				desiredContractionFactor = OMR_MIN(extensions->dnssExpectedTimeRatioMinimum  * 2, expectedTimeRatio);
-				desiredContractionFactor = desiredContractionFactor - _averageScavengeTimeRatio;
+					double desiredContractionFactor, adjustedContractionFactor;
```

Comment:
Existing code, but still... generally, try to always keep varaiable initialized from the vary declaration point and  move declarations as late as possible
Specifically:
- move declaration of desiredContractionFactor below when it's first used.
- for adjustedContractionFactor move it down closer just before the the first 'if' it's used, and init it with some neutral value (probably desiredContractionFactor, what will simplify the 'if' statement)